weather/src/js/api/weather-api.js

```diff
@@ -0,0 +1,80 @@
+import axios from 'axios';
+
+export function getWeatherData(units, coords) {
+    return new Promise(function(resolve, reject) {
+    
+    
+    axios.get('http://api.openweathermap.org/data/2.5/forecast',{
+        params : {
+                units: 'metric',
+                lat: coords.lat,
+                lon: coords.lon,
+                appid: '11ad38fb16a7a484238ba3e7396d2cf6'
+            }
+        })
+        .then(weatherData => {  
+            var list = new Array(5);//5days
+                
+                var periodlist= new Array(8); //8 periods
+                var currentHour = (new Date()).getHours();
+                var missingCols=  Math.ceil(currentHour / 3)==(currentHour / 3) ? (currentHour / 3) + 1 :Math.ceil(currentHour / 3);
+                var col = missingCols ;
+                for(var i=0;i<missingCols;i++){
+                    periodlist.push({ref:i});
+                }
+                weatherData.data.list.forEach( wdata => {
```

Comment:
I would suggest that you never use `forEach` as it indicates a block of code with side effects. Prefer `reduce` instead.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce?v=example